### ISSUE TITLE
Switch Workspace action doesn't work on GNOME 3.30

### ISSUE THREAD

**Issue by mpodshivalin** [mpiannucci/gnome-shell-extended-gestures#59]
I have this configuration:
![image](https://user-images.githubusercontent.com/16121630/47737576-20840580-dc69-11e8-876c-454b190f5cc9.png)

In GNOME 3.28 everything works as expected: when I swipe with 3 fingers down, I switch to a lower workspace
In GNOME 3.30, however, nothing happens.

Other actions work fine (I've tested Toggle Overview, Cycle Applications and Show App Drawer), Switch Workspace is the only action that doesn't work.

GNOME 3.30 has introduced a really smooth Switch Workspace gesture (maybe that is the root of the problem). You can swipe with 4 fingers up/down and switch between workspaces smoothly, similarly to macOS. I like this change, but I want to be able to swipe with 3 fingers instead of 4 to switch workspace (ideally smoothly).

Thank you for creating this extension, I've used it for a long time and I really like it!

**Comment by luoc0815** [comment]
I can confirm this issue for Fedora 29 as described in this thread: https://www.reddit.com/r/Fedora/comments/9sxjwk/getting_touch_gestures_to_work_in_fedora_29/

**Comment by mpiannucci** [comment]
Okay there are two different problems here: The first is that switch workspace action is not working with gnome 3.30. The fix for that is already pushed into the master branch. It has been reported before and they all found installing from master worked to fix it from them. 

The second is news to me and very interesting. I have not used gnome 3.30 but I willl try to install it this week. I use mac OS a lot so I am guessing what you mean by smoothly is that it tracks with your fingers as you swipe? Is that true?

**Comment by luoc0815** [comment]
> Okay there are two different problems here: The first is that switch workspace action is not working with gnome 3.30. The fix for that is already pushed into the master branch. It has been reported before and they all found installing from master worked to fix it from them.

Just installed from source on Fedora 29 and it works as expected. :)

> The second is news to me and very interesting. I have not used gnome 3.30 but I willl try to install it this week. I use mac OS a lot so I am guessing what you mean by smoothly is that it tracks with your fingers as you swipe? Is that true?

Seems like that's the case. It would be really great if you could add an option to map that to three finger gestures.

Here's an record of on how it looks like on Arch
https://www.reddit.com/r/gnome/comments/9hsw0p/nice_touchpad_gestures_improvement_in_gnome_330/

**Comment by mpodshivalin** [comment]
> Okay there are two different problems here: The first is that switch workspace action is not working with gnome 3.30. The fix for that is already pushed into the master branch. It has been reported before and they all found installing from master worked to fix it from them.
I've just installed from source on Fedora 29, everything works now, so the problem is solved

> The second is news to me and very interesting. I have not used gnome 3.30 but I willl try to install it this week. I use mac OS a lot so I am guessing what you mean by smoothly is that it tracks with your fingers as you swipe? Is that true?
Yes, it behaves exactly like macOS, which is a great improvement - I've waited a long time for this. It even works smoothly, as in macOS, even on 4K display. It would be great though to be able to switch with 3 fingers, not four.
BUT there is a bug which prevents me from enjoying it :) Basically if you go to overview, then use the gestures to switch workspaces (keyboard, mouse, etc works fine), and go out of overview, windows will not be displayed correctly. See more here:  https://gitlab.gnome.org/GNOME/gnome-shell/issues/644 (there is a video to demonstrate) 
And what's more interesting - I've just tested the issue when switching with 3 fingers gesture using this extension (so this is not a smooth gesture), and for some reason it  persists. I'm 99% sure this is GNOME problem, not extension's problem, but still

**Comment by mpiannucci** [comment]
Yup someone else reported the same thing with regard to the weird overwview while switching workspaces. On my radar now thank you!

**Comment by mpodshivalin** [comment]
I've just upgraded to Fedora 30 and GNOME Shell 3.32.
They've implemented smooth 4-finger gesture in the overview as well, and it doesn't cause any issues when switching.
But the extension still uses non-smooth scroll and there are still the same problems when switching workspaces in overview, now they are reproducible only with the extension and not within gnome shell itself.
